app/static/js/ajax-search.js

```diff
@@ -33,12 +36,27 @@ function ajaxCallSearch() {
 }
 
 $(function () {
-    $('#search').keydown(function () {
-        clearTimeout(requestTimer);
-        requestTimer = setTimeout(function () {
-            ajaxCallSearch();
-        }, 1000);
-    });
+    var params = new URLSearchParams(location.search.substring(1));
+    var search = params.get('search');
+    var defaultSearchBy = $("input[name=search_by]:checked").val();
+    var otherSearchBy = $("input[name=search_by]:not(:checked)").val();
+    var searchBy = params.get('search_by') === defaultSearchBy ? defaultSearchBy : otherSearchBy;
```

Comment:
We should be able to support more `search_by` parameters.

One thing I can think of is to generate an array of all the `search_by` parameters and validate with that, instead of just checking with a single default. This might be helpful if we plan to add more `search_by` params in the future.

Reply:
I do that only for security purpose.

To support multiple values, we could use instead:

```js
var availableSearchByValues = $("input[name=search_by]").get().map(function (el) { return el.value });
var searchBy = params.get('search_by');
if (availableSearchByValues.indexOf(searchBy) < 0) {
    searchBy = $("input[name=search_by]:checked").val();
}
```

Reply:
This looks better, it would allow for more flexibility. What possible security flaw do you see with it?

Reply:
It have a limited impact here, but [never trust user input](https://blog.sheddow.xyz/css-timing-attack/):

https://github.com/macports/macports-webapp/blob/8b5d68b08038a869ad54b62d87e8d17a3eb89f3a/app/static/js/ajax-search.js#L53

Reply:
I totally agree with you, but it should not be a problem here. However, I am fine with the current implementation also.